File: hyejin/Week_14/Daily Temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        
        int[] result = new int[temperatures.length];
        
        for(int i=0; i<temperatures.length-1; i++) {
            int count = 1;
            for(int j=i+1; j<temperatures.length; j++) {
                if(temperatures[j] > temperatures[i]) {
                    result[i] = count;
                    count = 0;
                    break;
                }    
                else
                    count++;
            
            }
        }
        return result;
    }
}

File: sangryeol/Week_14/Daily Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int size = temperatures.size();
        vector<int> answer(size, 0);
        stack<int> stk;
        vector<int>& T = temperatures;
        for (int i = size - 1; i >= 0; i--) {
            while (!stk.empty() && T[stk.top()] <= T[i]) stk.pop();
            if (!stk.empty()) answer[i] = stk.top() - i;
            stk.push(i);
        }
        return answer;
    }
};